### PR TITLE
Go-To Actions with transactions emails

### LINKED ISSUE #5671: Add go-to actions to transactional emails
It would be nice to have "Open document" and "Open comment" as one-click actions in our notification emails, should be straightforward to add the markup to trigger it wherever we have a CTA in the email itself.

https://developers.google.com/gmail/markup/reference/one-click-action

Could also have this for the login email.

### PR DESCRIPTION
**Pull request by apoorv-mishra**
Closes #5671

### CODE REVIEW

File: server/emails/mailer.tsx [hunk 1]
```diff
@@ -56,6 +56,63 @@ export class Mailer {
     }
   }
 
+  template = ({
```
Comment (tommoor) [inline server/emails/mailer.tsx:59]: In the end I'm not clear on why this customization is needed, it seems to just remove previewText and pass it into `EmailLayout` instead – but it seems unrelated to the right of the work, can you clarify?

File: server/emails/templates/BaseEmail.tsx [hunk 2]
```diff
@@ -149,6 +149,16 @@ export default abstract class BaseEmail<
    */
   protected abstract preview(props: S & T): string;
 
+  /**
+   * Returns the markup for the email message, email clients like Gmail
+   * use this markup to highlight important user actions
+   * within the inbox itself, like opening a document or a thread.
+   *
+   * @param props Props in email constructor
+   * @returns The preview text as a string
```
Comment (tommoor) [inline server/emails/templates/BaseEmail.tsx:158]: Comment incorrect

File: server/emails/templates/CollectionCreatedEmail.tsx [hunk 3]
```diff
@@ -57,6 +57,21 @@ export default class CollectionCreatedEmail extends BaseEmail<
     return `${collection.user.name} created a collection`;
   }
 
+  protected markup({ teamUrl, collection }: Props) {
```
Comment (tommoor) [inline server/emails/templates/CollectionCreatedEmail.tsx:60]: This has ended up pretty repetitive, it makes me want to abstract just a touch more – why not pass `potentialAction` to `EmailLayout` as a prop and have it construct the rest?

File: server/emails/mailer.tsx (hunk 1 above)
Comment (apoorv-mishra) [reply server/emails/mailer.tsx:59]: It puts `bodyContent` in the right place. [Refer the original template](https://github.com/revivek/oy/blob/master/src/utils/HTML4.js), which `bodyContent` further inside the nested structure. It's desirable for `bodyContent` to be the immediate child of `<body>` in our case since it contains the markup under `<script>`

File: server/emails/templates/CollectionCreatedEmail.tsx (hunk 3 above)
Comment (apoorv-mishra) [reply server/emails/templates/CollectionCreatedEmail.tsx:60]: Don't know, it sounded weird to pass the `potentialAction` as a prop, markup seemed more in-line, semantically. But can do, if you insist :smile:

Comment (tommoor) [reply server/emails/templates/CollectionCreatedEmail.tsx:60]: `goToAction` 👍

File: server/emails/mailer.tsx (hunk 1 above)
Comment (tommoor) [reply server/emails/mailer.tsx:59]: Can you add that link to the default template as a code comment, thanks

File: server/emails/templates/InviteAcceptedEmail.tsx [hunk 4]
```diff
@@ -60,7 +60,10 @@ Open ${env.APP_NAME}: ${teamUrl}
     unsubscribeUrl,
   }: Props & BeforeSendProps) {
     return (
-      <EmailTemplate>
+      <EmailTemplate
+        previewText={this.preview({ invitedName } as Props)}
+        goToAction={{ url: teamUrl, name: `Open ${env.APP_NAME}` }}
```
Comment (tommoor) [inline server/emails/templates/InviteAcceptedEmail.tsx:65]:
```suggestion
```

Doesn't make sense for this one

File: server/emails/templates/InviteEmail.tsx [hunk 5]
```diff
@@ -47,8 +47,13 @@ Join now: ${teamUrl}
   }
 
   protected render({ teamName, actorName, actorEmail, teamUrl }: Props) {
+    const inviteLink = `${teamUrl}?ref=invite-email`;
+
     return (
-      <EmailTemplate>
+      <EmailTemplate
+        previewText={this.preview()}
+        goToAction={{ url: inviteLink, name: "Join now" }}
```
Comment (tommoor) [inline server/emails/templates/InviteEmail.tsx:55]:
```suggestion
```

Doesn't make sense for this one, as you need to read the email first

File: server/emails/templates/InviteReminderEmail.tsx [hunk 6]
```diff
@@ -52,8 +52,12 @@ If you haven't signed up yet, you can do so here: ${teamUrl}
   }
 
   protected render({ teamName, actorName, actorEmail, teamUrl }: Props) {
+    const inviteLink = `${teamUrl}?ref=invite-reminder-email`;
     return (
-      <EmailTemplate>
+      <EmailTemplate
+        previewText={this.preview()}
+        goToAction={{ url: inviteLink, name: "Join now" }}
```
Comment (tommoor) [inline server/emails/templates/InviteReminderEmail.tsx:59]:
```suggestion
```

Doesn't make sense for this one, as you need to read the email first

File: server/emails/templates/WebhookDisabledEmail.tsx [hunk 7]
```diff
@@ -39,8 +39,12 @@ Webhook settings: ${teamUrl}/settings/webhooks
   }
 
   protected render({ webhookName, teamUrl }: Props) {
+    const webhookSettingsLink = `${teamUrl}/settings/webhooks`;
     return (
-      <EmailTemplate>
+      <EmailTemplate
+        previewText={this.preview({ webhookName } as Props)}
+        goToAction={{ url: webhookSettingsLink, name: "View Settings" }}
```
Comment (tommoor) [inline server/emails/templates/WebhookDisabledEmail.tsx:46]:
```suggestion
```

Doesn't make sense for this one, as you need to read the email first

File: server/emails/templates/WelcomeEmail.tsx [hunk 8]
```diff
@@ -44,8 +44,13 @@ ${teamUrl}/home
   }
 
   protected render({ teamUrl }: Props) {
+    const welcomLink = `${teamUrl}/home?ref=welcome-email`;
+
     return (
-      <EmailTemplate>
+      <EmailTemplate
+        previewText={this.preview()}
+        goToAction={{ url: welcomLink, name: `Open ${env.APP_NAME}` }}
```
Comment (tommoor) [inline server/emails/templates/WelcomeEmail.tsx:52]:
```suggestion
```

Doesn't make sense for this one, as you need to read the email first

File: server/emails/templates/CollectionCreatedEmail.tsx [hunk 9]
```diff
@@ -68,8 +68,13 @@ Open Collection: ${teamUrl}${collection.url}
   }
 
   protected render({ collection, teamUrl, unsubscribeUrl }: Props) {
+    const collectionLink = `${teamUrl}${collection.url}`;
+
     return (
-      <EmailTemplate>
+      <EmailTemplate
+        previewText={this.preview({ collection } as Props)}
+        goToAction={{ url: collectionLink, name: "View Collections" }}
```
Comment (tommoor) [inline server/emails/templates/CollectionCreatedEmail.tsx:76]:
```suggestion
        goToAction={{ url: collectionLink, name: "View Collection" }}
```

File: server/emails/templates/ExportSuccessEmail.tsx [hunk 10]
```diff
@@ -57,8 +57,13 @@ Your requested data export is complete, the exported files are also available in
   }
 
   protected render({ id, teamUrl, unsubscribeUrl }: Props & BeforeSendProps) {
+    const downloadLink = `${teamUrl}/api/fileOperations.redirect?id=${id}`;
+
     return (
-      <EmailTemplate>
+      <EmailTemplate
+        previewText={this.preview()}
+        goToAction={{ url: downloadLink, name: "Download" }}
```
Comment (tommoor) [inline server/emails/templates/ExportSuccessEmail.tsx:65]:
```suggestion
        goToAction={{ url: downloadLink, name: "Download export" }}
```

**Review by tommoor — APPROVED** [review]
Looks good, just don't need these actions on _every_ email.

**Review by tommoor — APPROVED** [review]
image
